server/internal/core/application/covenantless.go

```diff
@@ -38,12 +39,14 @@ type covenantlessService struct {
 	paymentRequests *paymentsMap
 	forfeitTxs      *forfeitTxsMap
 
-	eventsCh     chan domain.RoundEvent
+	eventsCh     chan interface{}
```

Comment:
I completely understand why the new "events" are not at the domain but at the application level since they contain ephemeral data that is unnecessary to persist.

That being said, if we define a `isEvent()` method for the new events we shouldn't need to make this change and we could keep this field as `chan domain.RoundEvent` and prevent other changes in the rest of the codebase